backend/api/v1/sql_service.go

```diff
@@ -1494,73 +1498,69 @@ func (s *SQLService) accessCheck(
 			case parserbase.SelectInfoSchema:
 				permission = iam.PermissionSQLInfo
 			case parserbase.Select:
-				// Conditional permission check below.
+				permission = iam.PermissionSQLSelect
 			default:
 			}
-			if isExplain {
-				permission = iam.PermissionSQLExplain
-			}
 			if span.Type == parserbase.DDL || span.Type == parserbase.DML {
 				if err := checkDataSourceQueryPolicy(ctx, s.store, s.licenseService, database, span.Type); err != nil {
 					return err
 				}
 			}
-			if permission != "" {
-				ok, err := s.iamManager.CheckPermission(ctx, permission, user, project.ResourceID)
-				if err != nil {
-					return connect.NewError(connect.CodeInternal, errors.Errorf("failed to check permission with error: %v", err.Error()))
-				}
-				if !ok {
-					return connect.NewError(connect.CodePermissionDenied, errors.Errorf("user %q does not have permission %q on project %q", user.Email, permission, project.ResourceID))
-				}
-			}
+		} else if span.Type == parserbase.Select {
+			permission = iam.PermissionSQLSelect
 		}
-		if span.Type == parserbase.Select {
-			if len(span.SourceColumns) == 0 {
-				if err := checkDatabaseAccess(); err != nil {
-					return err
-				}
+
+		if permission == "" {
+			// always fallback to bb.sql.select
+			permission = iam.PermissionSQLSelect
+		}
+
+		if len(span.SourceColumns) == 0 {
+			if err := checkDatabaseAccess(permission); err != nil {
+				return err
 			}
+		}
 
-			var deniedResources []string
-			for column := range span.SourceColumns {
-				attributes := map[string]any{
-					common.CELAttributeRequestTime:        time.Now(),
-					common.CELAttributeResourceDatabase:   common.FormatDatabase(instance.ResourceID, column.Database),
-					common.CELAttributeResourceSchemaName: column.Schema,
-					common.CELAttributeResourceTableName:  column.Table,
+		var deniedResources []string
+		for column := range span.SourceColumns {
```

Comment:
The refactoring changes the code behavior, for example:
Before: For explain statement, check the user has iam.PermissionSQLExplain in the project
Now: For explain statement, check the user has iam.PermissionSQLExplain on each column appeared in query

Not sure if bytebase support PermissionSQLExplain on column level